sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -389,6 +389,46 @@ case class WriteDelta(
   }
 }
 
+trait V2CreateTableAsSelectPlan extends Command with V2CreateTablePlan with KeepAnalyzedQuery {
```

Comment:
`KeepAnalyzedQuery` is for commands that need both analyzed and optimized plan. V2 write commands use it because if v1 fallback is not needed, we still need the optimized plan.

For CTAS/RTAS, we don't need the optimized plan at all. I think `AnalysisOnlyCommand` is a better fit here.

Reply:
Actually, we can remove `KeepAnalyzedQuery` now. The reason I added `KeepAnalyzedQuery` is because for CTAS/RTAS, we can only know if v1 fallback is needed or not at execution time when the table is created. But this problem goes away after this PR.

Reply:
Great point, `AnalysisOnlyCommand` already has the logic to hide children. Switched.

All other v2 write commands extend `KeepAnalyzedQuery`. It seems safe to remove it and use the optimized plan to construct `DataFrame` as that plan is marked as analyzed (so analysis won't be called again). We will re-optimize the plan after constructing `DataFrame` but it should be faster than optimizing only the analyzed plan (at least we can avoid planning scans multiple times). If there is no harm of using the optimized plan to create `DataFrame`, I'll follow up with a PR to remove `KeepAnalyzedQuery`.